arch/Kconfig

```diff
@@ -15,6 +15,7 @@ config ARCH_ARM
 	select ARCH_HAVE_STACKCHECK
 	select ARCH_HAVE_CUSTOMOPT
 	select ARCH_HAVE_STDARG_H
+	select ARCH_HAVE_SETJMP if !ARCH_TOOLCHAIN_IAR
```

Comment:
So both GCC and Clang need it yes?

Reply:
Yes, GCC and Clang can handle arch_setjmp.S correctly.